pyat/at/lattice/lattice_object.py

```diff
@@ -465,15 +448,17 @@ def reverse(self, copy=False):
             lattice(Lattice): reversed Lattice, if `copy==True` the
                               new lattice object is returned
                               otherwise None
+        Example:
+            >>> newring = ring.reverse(copy=True)
         """
+        elems = (el.swap_faces(copy=True) for el in reversed(self))
         if copy:
-            elems = (el.swap_faces(copy=copy) for el in self[::-1])
             return Lattice(elem_generator, elems, iterator=self.attrs_filter)
         else:
-            reversed_list = list(reversed(self))
+            reversed_list = list(elems)
```

Comment:
Doing this you copy the elements. Is it possible and would it make sense to return the elements themselves by using copy=copy in the elems generator?

Reply:
No copy is a problem: if the same element is used twice in the ring, it will be swapped twice, and the second one in the reversed lattice will be wrong.